src/client/Microsoft.Identity.Client/Platforms/iOS/Broker/iOSBroker.cs

```diff
@@ -271,17 +274,27 @@ private MsalTokenResponse ResultFromBrokerResponse(Dictionary<string, string> re
                         responseDictionary[iOSBrokerConstants.ApplicationToken]);
                 }
 
-                if (responseDictionary.ContainsKey(BrokerResponseConst.BrokerErrorCode) &&
-                    responseDictionary[BrokerResponseConst.BrokerErrorCode] == BrokerResponseConst.iOSBrokerUserCancellationErrorCode)
+                brokerTokenResponse = MsalTokenResponse.CreateFromiOSBrokerResponse(responseDictionary);
+                _accessToken = string.Empty;
+                _refreshToken = string.Empty;
+
+                if (responseDictionary.TryGetValue(BrokerResponseConst.BrokerErrorCode, out string errCode))
                 {
-                    responseDictionary[BrokerResponseConst.BrokerErrorCode] = MsalError.AuthenticationCanceledError;
+                    if(errCode == BrokerResponseConst.iOSBrokerUserCancellationErrorCode)
+                    {
+                        responseDictionary[BrokerResponseConst.BrokerErrorCode] = MsalError.AuthenticationCanceledError;
+                    }
+                    else if (errCode == BrokerResponseConst.iOSBrokerProtectionPoliciesRequired)
+                    {
+                        responseDictionary[BrokerResponseConst.BrokerErrorCode] = MsalError.ProtectionPolicyRequired;
+                        _accessToken = responseDictionary[BrokerResponseConst.AccessToken];
```

Comment:
It won't be enough to give MSAL just the AT and RT. You need to give it full response properties and ID token for caching like `CreateFromiOSBrokerResponse` does.